src/app/components/SearchForm.js

```diff
@@ -11,18 +12,39 @@ import {
     unsplashClient,
     storage
 } from './../constants';
+import type { InputEvent } from 'customLibrary';
+
+type PropsType = {
+    onSearch: Function,
+    clearQuery: string,
+    loadMore: Function
+}
+
+type StateType = {
+    inputValue: string,
+    buttonDisabled: boolean,
+    pageNumberToShow: number,
+    historyListIsActive: boolean,
+    historyList: Array<string>,
+    suggestionList: Array<any>
+}
 
-export default class SearchForm extends Component {
-    state = {
-        inputValue: '',
-        buttonDisabled: true,
-        pageNumberToShow: 1,
-        historyListIsActive: false,
-        historyList: storage.local,
-        suggestionList: []
-    };
-
-    componentWillReceiveProps(nextProps) {
+export default class SearchForm extends Component<PropsType, StateType> {
+    constructor(props: PropsType) {
+        super(props);
+        this.state = {
+            inputValue: '',
+            buttonDisabled: true,
+            pageNumberToShow: 1,
+            historyListIsActive: false,
+            historyList: storage.local,
+            suggestionList: []
+        };
+    }
+
+    state: StateType;
```

Comment:
not sure it should be specified like this, check please https://flow.org/en/docs/react/